namer/dnssrv/src/main/scala/io/buoyant/namer/dnssrv/DnsSrvNamer.scala

```diff
@@ -0,0 +1,76 @@
+package io.buoyant.namer.dnssrv
+
+import java.io.IOException
+import java.net.InetSocketAddress
+
+import com.twitter.finagle._
+import com.twitter.finagle.stats.StatsReceiver
+import com.twitter.logging.Logger
+import com.twitter.util.Activity.State
+import com.twitter.util._
+import org.xbill.DNS
+
+class DnsSrvNamer(prefix: Path, resolver: DNS.Resolver, timer: Timer, refreshInterval: Duration, stats: StatsReceiver)
+  extends Namer {
+
+  override def lookup(path: Path): Activity[NameTree[Name]] = memoizedLookup(path)
+
+  private val log = Logger.get("dnssrv")
+  private val memoizedLookup: (Path) => Activity[NameTree[Name]] = Memoize { path =>
+    path.take(1) match {
+      case id@Path.Utf8(address) =>
+        Activity(Var.async[State[NameTree[Name]]](Activity.Pending) { state =>
+          timer.schedule(refreshInterval) {
+            val next = lookupSrv(address, prefix ++ id, path.drop(1)) match {
+              case Return(nameTree) => Activity.Ok(nameTree)
+              case Throw(e) => Activity.Failed(e)
+            }
+            state.update(next)
+          }
+        })
+      case _ => Activity.value(NameTree.Neg)
+    }
+  }
+
+  private def lookupSrv(address: String, id: Path, residual: Path): Try[NameTree[Name]] = {
+    val question = DNS.Record.newRecord(
+      DNS.Name.fromString(address),
+      DNS.Type.SRV,
+      DNS.DClass.IN
+    )
+    val query = DNS.Message.newQuery(question)
+    log.debug("looking up %s", address)
+    Try(resolver.send(query)) flatMap { message =>
```

Comment:
I imagine that resolver.send is a potentially blocking call?  If so, then I think DnsSrvNamer should probably be constructed with a com.twitter.util.FuturePool so that these calls don't block the timer thread.

Furthermore, it may be a good idea to record a stat around the lookup times.